cmd/soroban-cli/src/commands/contract/read.rs

```diff
@@ -179,10 +179,10 @@ impl NetworkRunnable for Cmd {
 
     async fn run_against_rpc_server(
         &self,
-        _: Option<&global::Args>,
-        config: Option<&config::Args>,
+        _global_args: Option<&global::Args>,
+        _config: Option<&config::Args>,
     ) -> Result<FullLedgerEntries, Error> {
-        let config = config.unwrap_or(&self.config);
+        let config: config::Args = self.config.clone().into();
```

Comment:
```suggestion
        let config = self.config;
```
You don't seem to need to use the bigger config so why even convert when you have access to the network and locator?